src/components/data-management/CellLevelUploadModal.jsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import {
+  Modal,
+  Button,
+  Typography,
+  Space,
+  Row,
+  Col,
+  Empty,
+  Divider,
+  List,
+} from 'antd';
+import { CheckCircleTwoTone, DeleteOutlined } from '@ant-design/icons';
+import Dropzone from 'react-dropzone';
+
+import handleError from 'utils/http/handleError';
+import endUserMessages from 'utils/endUserMessages';
+import readFileToString from 'utils/upload/readFileToString';
+
+const { Text, Title, Paragraph } = Typography;
+
+const CellLevelUploadModal = (props) => {
+  const { onUpload, onCancel } = props;
+
+  const [canUpload, setCanUpload] = useState(false);
+  const [filesList, setFilesList] = useState([]);
+
+  useEffect(() => {
+    setCanUpload(filesList.length === 1);
+  }, [filesList]);
+
+  // Handle on Drop
+  const onDrop = async (droppedFiles) => {
+    if (droppedFiles.length !== 1 || !droppedFiles[0].name.endsWith('.tsv')) {
+      handleError('error', endUserMessages.ERROR_METADATA_MULTIPLE_FILES);
+      return;
+    }
+
+    const file = droppedFiles[0];
+    const data = await readFileToString(file);
+
+    if (!data.trim().split('\n')[0].trim().split('\t').includes('barcode')) {
+      handleError('error', endUserMessages.ERROR_CELL_LEVEL_COLUMN);
+      return;
+    }
+    setFilesList([file]);
+  };
+
+  const removeFile = (fileName) => {
+    const newArray = _.cloneDeep(filesList);
+    const fileIdx = newArray.findIndex((file) => file.name === fileName);
+    newArray.splice(fileIdx, 1);
+    setFilesList(newArray);
+  };
+
+  return (
+    <Modal
+      title=''
+      visible
+      onCancel={onCancel}
+      width='50%'
+      footer={(
+        <Button
+          type='primary'
+          key='create'
+          block
+          disabled={!canUpload}
+          onClick={() => {
+            onUpload(filesList[0]);
+            setFilesList([]);
+          }}
+        >
+          Upload
+        </Button>
+      )}
+    >
+      <Row style={{ margin: '1rem 0' }}>
+        <Col span={24}>
+          <Title level={4}>
+            File upload Upload:
+            <span style={{ color: 'red', marginRight: '2em' }}>*</span>
+          </Title>
+          <Paragraph>
+            Upload a single file containing the metadata in key-value tab-separated format (.tsv)
```

Comment:
This isn't the whole help text https://docs.google.com/document/d/1YM43BaBYMKjg-Q9GbgDk0AyNnTgNwpIVif89utePu50/edit?disco=AAAA4HPstSY